tests/src/testassembler.cpp

```diff
@@ -81,7 +81,7 @@ auto SimpleAssemblersTest(const PreBasis& preBasis) {
     Dune::FieldVector<double, 2> pos{2.0, 1.0};
     int centerNode = 1; // number of nodes at the center ( = pos)
     if (Ikarus::Concepts::LagrangeNodeOfOrder<ChildType, 1> and ref == 0) {
-      t.checkThrow<Dune::GridError>(
+      t.checkThrow<Dune::InvalidFutureException>(
```

Comment:
@tarun-mitruka I overlooked that Exception name. I think this is also not a useful exception type here.

Reply:
maybe we have to create our own Exceptions anyway by deriving from https://gitlab.dune-project.org/core/dune-common/-/blob/master/dune/common/exceptions.hh?ref_type=heads

Reply:
In the [code](https://github.com/ikarus-project/ikarus/blob/fafb16a98574b96f0d449b1b1a2658d28213c3fb/ikarus/utils/functionhelper.hh#L75) it is `InvalidStateException` and here it is different, but nevertheless there is nothing nice that fits correctly here may be. We can create our own exceptions in a separate PR, and use anything here

Reply:
I think this is a `InvalidStateExecption`, I mistyped it here i think

Reply:
> In the [code](https://github.com/ikarus-project/ikarus/blob/fafb16a98574b96f0d449b1b1a2658d28213c3fb/ikarus/utils/functionhelper.hh#L75) it is `InvalidStateException` and here it is different, but nevertheless there is nothing nice that fits correctly here may be. We can create our own exceptions in a separate PR, and use anything here

Thats fine